tools/mksurfdata_map/src/mksurfdat.F90

```diff
@@ -1507,6 +1482,7 @@ subroutine normalizencheck_landuse(ldomain)
        end if
        
        ! Roundoff error fix
+       ! KO - Not sure if this needs to be changed or is necessary
```

Comment:
Regarding your first comment on the rounoff error check, specifically about setting the 25s to 20s or even something a bit lower.  I've created some offline double-precision code to understand this.
If I set each of the five landunits (pctlak, pctwet, pcturb, pctgla, pctcrop) to something like 20.d - 1.d-**7** one at a time, then the code seems to correctly fix this such that suma = 100.  
If I set more than one of the landunits (up to four of them simultaneously) to 20.d - 1.d-**7**, then the code also seems to correctly fix this as well.
If I set all five landunits to 20.d - 1.d-**7** then I get the abort error.
I can fix this by setting the threshold in the if statements to 20.d - 1.d-**6**.
So I'm wondering if this is the situation you had in mind.

Reply:
> Regarding your first comment on the rounoff error check, specifically about setting the 25s to 20s or even something a bit lower. I've created some offline double-precision code to understand this.
> If I set each of the five landunits (pctlak, pctwet, pcturb, pctgla, pctcrop) to something like 20.d - 1.d-7 one at a time, then the code seems to correctly fix this such that suma = 100.
> If I set more than one of the landunits (up to four of them simultaneously) to 20.d - 1.d-7, then the code also seems to correctly fix this as well.
> If I set all five landunits to 20.d - 1.d-7 then I get the abort error.
> I can fix this by setting the threshold in the if statements to 20.d - 1.d-6.
> So I'm wondering if this is the situation you had in mind.

The main situation I had in mind is where we set `new_total_natveg_pct = 100._r8 - suma` and `suma` is essentially 100 but not exactly, so we end up setting `new_total_natveg_pct` to roundoff-level greater than 0. Then the main thing we need to do is to correct the natural veg pct to be exactly 0. But I can see why maybe it's a good idea to add the difference back in to some other landunit in this situation (as is done in the current code), e.g., to handle the situation where the difference from 100 is a bit greater than roundoff (and so a bit greater than what we typically allow).

My rationale for changing the 25 to 20 is indeed a scenario like you described above: It seems like the intent here is to find one of the largest other landunits to which we can add the difference. But now that we are including crop here as well, it's possible that none of crop, wetland, lake, glacier or urban would exceed 25% (e.g., if all of them have 20% cover). And yes, exactly as you say, I was worried about a theoretical situation where all of the other landunits have areas just a tiny bit below 20%.

Since it doesn't seem very important which landunit is picked here (after all, it looks like at most we're adding something like 1e-6 to one of the landunits), it seems like all you really need to do is find some landunit with a pct cover greater than 1% or so, and add the tiny remainder to it: I don't think it really matters if we add the remainder to a landunit with area 1% or 30%. So my main concern here is to not accidentally abort when things are really okay.

Reply:
I've set the threshold to 1% everywhere and left the landunit order the same as before.  If lake is greater than or equal to 1% then the adjustment will be added to it, etc.

Reply:
Regarding the second error check ("about 25 lines down from here"), I decided to remove the part involving epsilon but keep the last part that checks whether the sum of landunits is within 1e-10 of 100%.  I could have reworked the first part to include crop, but was unsure if the 4*epsilon(suma) should be changed to (5*epsilon(suma)?).  The last part seems useful as a final check to make sure landunits add to 100%.